src/test/java/edu/umich/lib/dor/replicaexperiment/ReplicationTest.java

```diff
@@ -0,0 +1,176 @@
+package edu.umich.lib.dor.replicaexperiment;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import edu.umich.lib.dor.replicaexperiment.domain.InfoPackage;
+import edu.umich.lib.dor.replicaexperiment.domain.Replica;
+import edu.umich.lib.dor.replicaexperiment.domain.Repository;
+import edu.umich.lib.dor.replicaexperiment.exception.NoEntityException;
+import edu.umich.lib.dor.replicaexperiment.service.InfoPackageService;
+import edu.umich.lib.dor.replicaexperiment.service.OcflFilesystemRepositoryClient;
+import edu.umich.lib.dor.replicaexperiment.service.ReplicaService;
+import edu.umich.lib.dor.replicaexperiment.service.Replication;
+import edu.umich.lib.dor.replicaexperiment.service.ReplicationFactory;
+import edu.umich.lib.dor.replicaexperiment.service.RepositoryClientRegistry;
+import edu.umich.lib.dor.replicaexperiment.service.RepositoryService;
+
+public class ReplicationTest {
+    InfoPackageService packageServiceMock;
+    RepositoryService repositoryServiceMock;
+    ReplicaService replicaServiceMock;
+    RepositoryClientRegistry registryMock;
+    InfoPackage infoPackageMock;
+    Path stagingPath;
+
+    OcflFilesystemRepositoryClient sourceClientMock;
+    OcflFilesystemRepositoryClient targetClientMock;
+    Replica replicaMock;
+    Repository sourceRepositoryMock;
+    Repository targetRepositoryMock;
+
+    ReplicationFactory replicationFactory;
+
+    @BeforeEach
+    void init() {
+        this.packageServiceMock = mock(InfoPackageService.class);
+        this.repositoryServiceMock = mock(RepositoryService.class);
+        this.replicaServiceMock = mock(ReplicaService.class);
+        this.registryMock = mock(RepositoryClientRegistry.class);
+        this.infoPackageMock = mock(InfoPackage.class);
+        this.stagingPath = Paths.get("staging");
+
+        this.replicationFactory = new ReplicationFactory(
+            packageServiceMock,
+            repositoryServiceMock,
+            replicaServiceMock,
+            registryMock,
+            stagingPath
+        );
+
+        this.sourceClientMock = mock(OcflFilesystemRepositoryClient.class);
+        this.targetClientMock = mock(OcflFilesystemRepositoryClient.class);
+        this.replicaMock = mock(Replica.class);
+        this.sourceRepositoryMock = mock(Repository.class);
+        this.targetRepositoryMock = mock(Repository.class);
```

Comment:
Could I just be using `Repository` or `InfoPackage` POJOs, or is it better to use mocks?